[19.0][MIG] product_pricelist_revision: Migration to 19.0

BhaveshHeliconia opened the pull request.

sergio-teruel committed:
11.0[WIP]product_pricelist_revision

sergio-teruel committed:
[11.0][IMP] product_pricelist_revision: Complete translations

ernestotejeda committed:
[MIG] product_pricelist_revision: Migration to 12.0

[UPD] Update product_pricelist_revision.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_pricelist_revision
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_pricelist_revision/

sergio-teruel committed:
[MIG] product_pricelist_revision: Migration to v13.0

sergio-teruel committed:
[IMP] product_pricelist_revision: black, isort

carlosdauden committed:
[IMP] product_pricelist_revision: Add percent price column and current filter
TT28144

[UPD] Update product_pricelist_revision.pot

claudiagn committed:
Added translation using Weblate (Catalan)

claudiagn committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (30 of 30 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_pricelist_revision
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_pricelist_revision/ca/

CarlosRoca13 committed:
[FIX] product_pricelist_revision: Applied the new way for digits
Before this change on runbot we caught a Warning about deprecated call to decimal_precision.get_precision(<application>)

[UPD] README.rst

product_pricelist_revision 13.0.1.1.0

carlosdauden committed:
[IMP] product_pricelist_revision: Allow create pricelist item

TT27524

joao-p-marques committed:
[IMP] *: pre-commit execution

Fix website key in the manifests

victoralmau committed:
[MIG] product_pricelist_revision: Migration to 14.0
TT31815

sebalix committed:
[FIX] Fix access right issues when using actions

sebastienbeau committed:
product_pricelist_revision: fix date_end/date_start are datetime

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 31.0% (9 of 29 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_pricelist_revision
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_pricelist_revision/it/

CarlosRoca13 committed:
[MIG] product_pricelist_revision: Migration to 15.0

Yadier-Tecnativa committed:
[MIG] product_pricelist_revision: Migration to 16.0

josep-tecnativa committed:
[IMP] product_pricelist_revision: test performance improvement
- Include context keys for avoiding mail operations overhead.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_revision
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_revision/it/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (29 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_revision
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_revision/fr/

Commit:
[IMP] product_pricelist_revision: pre-commit auto fixes

Commit:
[MIG] product_pricelist_revision: Migration to 17.0

BhaveshHeliconia committed:
[MIG] product_pricelist_revision: Migration to 18.0

Commit:
[UPD] Update product_pricelist_revision.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[MIG] product_pricelist_revision: Migration to 19.0

BhaveshHeliconia committed:
[MIG] product_pricelist_revision: Migration to 19.0

rousseldenis commented:
/ocabot migration product_pricelist_revision

gdgellatly reviewed (approved):
Functional test on own db plus code review LGTM.

gdgellatly reviewed (commented):
Sorry I put review against wrong module. I now tested this one on live database and it didn't work. It duplicated the item just fine but did not take the variance. Will get back to tomorrow.

BhaveshHeliconia commented:
> Sorry I put review against wrong module. I now tested this one on live database and it didn't work. It duplicated the item just fine but did not take the variance. Will get back to tomorrow.

Hi @gdgellatly,

I have tested this module locally and on Runbot, and it is working as expected.

Please note that this module's duplicate wizard only works with fixed price items (compute_price = 'fixed'). It does not apply the variance for other pricing types such as formula or percentage/discount. The wizard view also mentions this:

"Note: Only works with fixed price items. The variation % will be applied to calculate the new price."

If you are facing any issue specifically with fixed price items, could you please elaborate on the steps to reproduce and where exactly you are getting stuck? That would help us investigate further.

gdgellatly commented:
> > Sorry I put review against wrong module. I now tested this one on live database and it didn't work. It duplicated the item just fine but did not take the variance. Will get back to tomorrow.
> 
> Hi @gdgellatly,
> 
> I have tested this module locally and on Runbot, and it is working as expected.
> 
> Please note that this module's duplicate wizard only works with fixed price items (compute_price = 'fixed'). It does not apply the variance for other pricing types such as formula or percentage/discount. The wizard view also mentions this:
> 
> "Note: Only works with fixed price items. The variation % will be applied to calculate the new price."
> 
> If you are facing any issue specifically with fixed price items, could you please elaborate on the steps to reproduce and where exactly you are getting stuck? That would help us investigate further.

Hi @BhaveshHeliconia, yes sorry we worked that out and deleted the module. We have no use case for it so will remove my review. EDIT: I no longr have access to remove reviews so we will just have to live with it, it did work, just not how expected.

CarlosRoca13 committed:
[FIX] product_pricelist_revision: Fix date range in duplicate wizard test

Odoo 18 enforces a strict date_end > date_start constraint on pricelist items. The test was using the same datetime for both fields, causing a ValidationError.

See https://github.com/odoo/odoo/blob/36a7cadbfb9007ebddd660b7709e4bef58e96c2d/addons/product/models/product_pricelist_item.py#L307

CarlosRoca13 reviewed (commented):
Please add this change: https://github.com/OCA/product-attribute/pull/2291

BhaveshHeliconia commented:
> Please add this change: #2291

Done, thanks!

pedrobaeza commented:
/ocabot merge nobump
